File: 2581번： 소수/소수.c
/* ************************************************************************** */
/*                                                                            */
/*                                                      :::    :::    :::     */
/*   Problem Number: 2581                              :+:    :+:      :+:    */
/*                                                    +:+    +:+        +:+   */
/*   By: luchael <boj.kr/u/luchael>                  +#+    +#+          +#+  */
/*                                                  +#+      +#+        +#+   */
/*   https://boj.kr/2581                           #+#        #+#      #+#    */
/*   Solved: 2025/02/25 14:32:38 by luchael       ###          ###   ##.kr    */
/*                                                                            */
/* ************************************************************************** */
#include <stdio.h>
#include <math.h>
int main()
{
    int i, j, k, sum[2] = {0, 0};
    scanf("%d %d", &i, &j);
    for (; i <= j; i++)
    {
        if (i == 1)
            continue;
        for (k = round(sqrt((double)i)); k >= 1; k--)
        {
            if (!(k - 1))
            {
                sum[1] += i;
                if (!sum[0])
                    sum[0] = i;
            }
            if (!(i % k))
                break;
        }
    }
    if (sum[0])
        printf("%d\n%d", sum[1], sum[0]);
    else
        printf("%d", -1);
}

File: 2581번： 소수/소수.py
#  **************************************************************************  #
#                                                                              #
#                                                       :::    :::    :::      #
#    Problem Number: 2581                              :+:    :+:      :+:     #
#                                                     +:+    +:+        +:+    #
#    By: luchael <boj.kr/u/luchael>                  +#+    +#+          +#+   #
#                                                   +#+      +#+        +#+    #
#    https://boj.kr/2581                           #+#        #+#      #+#     #
#    Solved: 2025/02/25 14:33:08 by luchael       ###          ###   ##.kr     #
#                                                                              #
#  **************************************************************************  #
i = int(input())
j = int(input())
s = [0, 0]
for i in range(i,j+1):
    if i==1: continue
    for k in range(round(i**0.5), 0, -1):
        if (k-1)==0:
            s[1]+=i
            if s[0]==0: s[0] = i
        if i%k==0: break

if s[0]!=0:
    print(s[1])
    print(s[0])
else: print(-1)